src/mds/MDSAuthCaps.h

```diff
@@ -300,17 +299,23 @@ class MDSAuthCaps
     }
   }
 
-  bool root_squash_in_caps() const {
-    for (const MDSCapGrant &g : grants) {
-      if (g.match.root_squash) {
-        return true;
+  bool root_squash_in_caps(std::string_view fs_name) const {
+    for (const MDSCapGrant& g : grants) {
+      if (fs_name_match(g.match.fs_name, fs_name)) {
+        if (g.match.root_squash) {
+          return true;
+        }
       }
     }
     return false;
   }
 
   friend std::ostream &operator<<(std::ostream &out, const MDSAuthCaps &cap);
   std::string to_string();
+protected:
+  static bool fs_name_match(std::string_view capfs, std::string_view other) {
+    return capfs == other || capfs.empty() || capfs == "*";
```

Comment:
`capfs` sounds confusing to me. Like elsewhere in `MDSAuthCaps.cc` and `MDSAuthCaps.h`, `fs_name` is better IMO since it clearly describes that it will hold name of a FS.

And `fsname` is even better than `fs_name`  because that is also how we mention FS name in MDS caps.

Reply:
Check now.

Reply:
Looks great!